addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -1223,3 +1223,86 @@ def test_pos_payment_method_copy(self):
         self.assertFalse(pm_3.journal_id)
         self.assertTrue(pm_4)
         self.assertEqual(pm_4.journal_id.type, "bank")
+
+    def test_single_config_global_invoice(self):
+        """For a single POS config, create multiple orders and consolidate them into a single invoice"""
+        self.open_new_session()
+        # create orders
+        orders = []
+        orders.append(self.create_ui_order_data(
+            [(self.product1, 2), (self.product4, 3)],
+            payments=[(self.bank_pm1, 49.88)]
+        ))
+        orders.append(self.create_ui_order_data(
+            [(self.product4, 1), (self.product2, 5)],
+            payments=[(self.bank_pm1, 109.96)]
+        ))
+
+        # sync orders
+        self.env['pos.order'].sync_from_ui(orders)
+        # close the session
+        self.pos_session.action_pos_session_validate()
+
+        pos_orders = self.env['pos.order'].search([])
+        # set customer for the orders
+        pos_orders.write({'partner_id': self.customer.id})
+
+        # create consolidated invoice
+        self.env['pos.make.invoice'].create({"consolidated_billing": True}).with_context({"active_ids": pos_orders.ids}).action_create_invoices()
+        # check if have single invoice
+        self.assertEqual(len(pos_orders.account_move), 1)
+        self.assertEqual(pos_orders.account_move.partner_id, self.customer)
+        self.assertEqual(pos_orders.account_move.amount_total, sum(pos_orders.mapped('amount_total')))
+        self.assertEqual(pos_orders.account_move.state, 'posted')
+
+    def test_multi_config_global_invoice(self):
+        self.open_new_session()
+        orders = []
+        orders.append(self.create_ui_order_data(
+            [(self.product1, 3), (self.product2, 10)],
+            payments=[(self.bank_pm1, 230)]
+        ))
+        orders.append(self.create_ui_order_data(
+            [(self.product1, 5), (self.product0, 10)],
+            payments=[(self.bank_pm1, 50)]
+        ))
+        self.env['pos.order'].sync_from_ui(orders)
+        self.pos_session.action_pos_session_validate()
+
+        # creation of Config 2
+        # open_new_session will create session of config present in self.config
+        config2 = self.env['pos.config'].create({
+            'name': 'Main 2',
+            'journal_id': self.config.journal_id.id,
+            'invoice_journal_id': self.company_data['default_journal_sale'].id,
+            'payment_method_ids': [(4, self.bank_pm1.id)]
+        })
+        self.config = config2
```

Comment:
no. the 2nd session should come from the original config.

there should just be 1 generated invoice from the orders coming from different sessions.